debian/dpl4hydra.1
.TH "DPL4HYDRA" "1" "19/01/2014"
.SH NAME
DPL4HYDRA \- Generates a (d)efault (p)assword (l)ist as input for THC hydra
.SH DESCRIPTION
This script allow generate list with the default passwords of of the main 
vendors of network devices (BRAND) like: 2wire, 360systems, 3com, 3m, acc, 
acceleratednetworks, accton, aceex, acer, actiontec, adaptecraid, adc, 
adckentrox, adcompleteco, adic, adp, adtech, adtran, advanced, advanteknetworks,
aethra, airlink, aironet, aladdin, alcatel, allied, allnet, allot, alteon, ambit,
ami, amitech, amptron, amptron, andovercontrols, aoc, apache, apc, apple, areca, 
arescom, arlotto, arrowpoint, asante, ascend, ascom, asmack, asmax, aspect, ast,
asus, at&t, atcom, atlantis, att, attachmate, audioactive, autodesk, avaya,
avengernewssystem, award, axis, aztech, barracudanetworks, baynetworks, bea,
beetal, belkin, bestpractical, betabrite, billion, bintec, biodata, biostar,
bizdesign, blackbox, bluecoatsystems, bmc, borland, boson, boson, breezecom,
broadlogic, brocade, brother, bt, buffalo, cableandwireless, cabletron, canyon,
castlenet, cayman, celerity, cgi, cgiworld, chase, chaseresearch, checkpoint, 
chumingchen, ciphertrust, cisco, cnet, cobalt, colubrisnetworks, comersus,
compaq, compex, compualynx, computer, comtrend, conceptronic, concord, conexant,
conitec, corecess, coyotepoint, crystalview, ctx, cyberguard, cybermax, cyclades,
d-link, daewoo, dallas, dallassemiconductors, darkman, data, datacom, datageneral,
datawizard, davox, daytek, debian, decnet, deerfield, dell, demarc, deutschetelekom,
develcon, dictaphone, digicom, digicorp, digicraftsoftware, digiinternational, digital,
digitalequipment, discar, dlink, draytek, drupal.org, dupont, dynalink, dynix, econ, edimax,
efficientnetworks, elron, elsa, eminent, encad, enhydra, enox, enterasys, entrust, epox,
ericsson, etech, everfocus, exindanetworks, extendedsystems, extremenetworks, f5, firebird,
flowpoint, fortinet, foundry, foundrynetworks, freetech, fujixerox, funk, funksoftware,
galacticomm, gandalf, gateway, ge, geeklog, generalinstruments, gericom, giga, gigabyte,
globespanvirata, gossamerthreads, grandstreamnetworks, greatspeed, guardone, h2o, h2oproject,
harris, hayes, hewlett-packard, honeywell, horizondatasys, hosting, hp, huawei, ibm, 
imperiasoftware, informix, infosmart, infotec, infrant, innovaphone, integratednetworks,
intel, interbase, intermec, intershop, intex, intuit, ipstar, ipswitch, irc, ironport, iso, 
iwill, jd, jdedwards, jds, jetform, jetway, johnsoncontrols, josstechnology, juniper, justin,
justinhagstrom, kalatel, kaptest, kethinov, keyscan, konicaminolta, kti, kyocera, lacie, lanier,
lantronics, leading, lenel, level1, lg, lgic, linksys, linux, livingstone, logitech, longshine, 
lucent, machspeed, macromedia, magic-pro, main, mambo, mantis, marconi, mcafee, mediatrix, megastar, 
memotec, mentec, mercury, metro, michiel, microcom, micron, micronet, micronics, microplex, microrouter, 
microsoft, mikepeters, mikrotik, minoltaqms, mintel, mitel, mobotix, motorola, mp3mystic, mro, mrv, 
multitech, mutaresoftware, muze, mysql, nai, nanoteq, ncr, netapp, netcomm, netgear, netgenesis, netopia,
netport, netscape, netscreen, network, networkassociates, networkice, nexland, next, ngsec, ngsecure, 
nicesystemsltd, nimble, no, nokia, norstar, nortel, novell, nrg, nullsoft, nurit, oce, ods, oki, olicom,
olitec, omnitronix, oodiecom, openconnect, openmarket, openwave, openxchange, optivision, optus, oracle,
osicom, ovislink, pacific, packardbell, panasonic, pandatel, patton, penril, pentagram, pentaoffice, 
pentasafe, philips, phoenix, phpreactor, phptest, pirelli, planet, pollsafe, polycom, postgresql, 
prestige, prestigio, primebase, prolite, promise, prostar, proxim, pyramidcomputer, qdi, quake,
qualiteam, quantex, quantum, radio, radioshack, radware, rainbow, rampnetworks, rapidstream,
raritan, raytalk, redhat, redline, remedy, research, resumix, ricoh, rizen, rm, roamabout, rodopi,
safecom, sagem, sambartechnologies, samsung, samuelabels, sap, sap, schneider, schneiderelectric,
scientificatlanta, seagullscientific, securicor3net, securstar, semaphore, servertechnology, sgi,
sharp, shiva, shoretel, shuttle

This script depends on a local (d)efault (p)assword (l)ist called
~/.dpl4hydra/dpl4hydra_full.csv. If it is not available, regenerate it with
dpl4hydra refresh. Source of the default password list is http://open-sez.me/passwd.htm
.SH Options
.TP
.B help
Show help information
.TP
.B refresh
Refresh list: Download the full (d)efault (p)assword (l)ist
.TP
.B BRAND
Generates a (d)efault (p)assword (l)ist from the local file
~/.dpl4hydra/dpl4hydra_full.csv, limiting the output to BRAND systems, using
the format username:password (as required by THC hydra).

Dump list of brand selected in ~/.dpl4hydra/dpl4hydra_BRAND.lst

.TP
.B all
Dump list of all systems credentials into ~/.dpl4hydra/dpl4hydra_all.lst

.SH EXAMPLES
.BR
.nf 
dpl4hydra linksys
.fi
.nf
Output:
.fi
File dpl4hydra_linksys.lst was created with 20 entries
.nf
For use in hydra:
.fi
.nf
hydra \-C ~/.dpl4hydra/dpl4hydra_linksys.lst \-t 1 192.168.1.1 http\-get /index.asp
.fi

.SH SEE ALSO
.BR hydra (1),
.br
.SH AUTHOR
dpl4hydra was written by Roland Kessler.

.PP
This manual page was written by Daniel Echeverry <epsilon77@gmail.com>,
for the Debian project (and may be used by others).



debian/changelog
hydra (8.1-1) unstable; urgency=medium

  * New upstream release.
  * Bump Standards-Version to 3.9.6 (no changes).
  * Refresh 02_using_supported_protocols.diff and
    08_fix_spelling_mistakes.diff patches.
  * Remove d/source/options, now xz format is default.
  * Update 08_fix_spelling_mistakes.diff patch.
  * d/copyright: Add missing License values.

 -- Julián Moreno Patiño <julian@debian.org>  Wed, 10 Dec 2014 07:33:34 -0500

hydra (8.0-1) unstable; urgency=medium

  * New upstream release. (LP: #1275463)
  * debian/control
    + Bump Standards-Version to 3.9.5 (no changes).
    + Add dh-exec in B-D
    + Add wget | curl to Recommends field
  * debian/patches
    + Refresh all patches
    + Add 09_use_correct_install_dir_dlp4hydra.diff
      + Change upstream path for one suitable for GNU Debian
  * debian/copyright
    + Extend copyright holders years
    + Add dpl4hydra.sh copyright 
  * Remove debian/source/options
    + Use xz compression.
  * debian/rules
    + Remove override_dh_builddeb
      + Now dpkg-deb uses xz compression by default
  * debian/hydra.manpages
    + Add dpl4hydra.1 file
    + Add hydra-wizard.1 file
  * Add hydra.install file
    + Include two new scripts
      + dpl4hydra: Generate a default password list
      + hydra-wizard: A wizard script to use Hydra

 -- Daniel Echeverry <epsilon77@gmail.com>  Fri, 30 May 2014 22:05:44 -0500

hydra (7.5-2) unstable; urgency=low

  * debian/control
    + Remove libncp-dev in B-D. (Closes: #728864)
  * debian/patches
    + Refresh 02_using_supported_protocols.diff patch.
      + Remove the support of ncp protocol.

 -- Daniel Echeverry <epsilon77@gmail.com>  Fri, 15 Nov 2013 22:02:21 -0500

hydra (7.5-1) unstable; urgency=low

  * New upstream release
  * debian/copyright
    + Move the license from GPL-3.0+ to AGPL-3.0 with OpenSSL exception
    + Extend copyright holders years
  * debian/patches
    + Refreshed patch 02_using_supported_protocols.diff
    + Refreshed patch 05_add_option_to_cflag.diff
    + Refreshed patch 07_remove_troubled_files.diff
    + Refreshed patch 08_fix_spelling_mistakes.diff

 -- Daniel Echeverry <epsilon77@gmail.com>  Sat, 31 Aug 2013 22:28:57 -0500

hydra (7.4.2-2) unstable; urgency=low

  * Use ${binary:Version} in hydra-gtk package
    instead bump manually the version.
  * Disable libafpclient-dev support due to #702889
    + Update 02_using_supported_protocols.diff patch to remove
      the support of afp protocol.
  * Use canonical URIs for Vcs-* fields.
  * Add Keywords entry to d/hydra-gtk.desktop.

 -- Julián Moreno Patiño <julian@debian.org>  Mon, 15 Jul 2013 15:44:45 -0500

hydra (7.4.2-1) unstable; urgency=low

  * New upstream release.

 -- Daniel Echeverry <epsilon77@gmail.com>  Sat, 12 Jan 2013 09:23:51 -0500

hydra (7.4-1) unstable; urgency=low

  * New upstream release.
  * d/watch: Update regular expression.
  * d/control:
    + Remove DM-Upload-Allowed.
    + Update my email.
    + Add libncurses-dev to B-D.
    + Improve description. Thanks to Justin B Rye. (Closes: #680540)
    + Bump Standards-Version to 3.9.4 (no changes).
  * d/copyright: Update my email.
  * d/patches:
    + Refresh 02_using_supported_protocols.diff patch offset.
    + Update 04_fix_hyphen-used-as-minus-sign.diff.
    + Refresh 05_add_option_to_cflag.diff patch offset.
    + Add 08_fix_spelling_mistakes.diff patch to fix spelling
      mistakes.

 -- Julián Moreno Patiño <julian@debian.org>  Sun, 23 Dec 2012 14:22:46 -0500

hydra (7.3-1) unstable; urgency=low

  * New upstream release.
  * Remove 04_fix_spelling_error.diff patch
    + Merge with upstream.
  * Remove some lines in 05_add_option_to_cflag.diff patch.
    + Merge with upstream.
    + Improve DEP3 Description.
  * Refresh 07_remove_troubled_files.diff patch offset.
  * Add 04_fix_hyphen-used-as-minus-sign.diff patch to
    fix hyphen-used-as-minus-sign lintian warning in the
    manpage.

 -- Julián Moreno Patiño <darkjunix@gmail.com>  Wed, 23 May 2012 21:25:24 -0500

hydra (7.2-2) unstable; urgency=low

  * Update 05_add_option_to_cflag.diff patch to use properly
    Hardening flags. (Closes: #665281)
    + Thanks to Simon Ruderich for the patch.
  * Bump debhelper to 9 in B-D.
    + Remove debian/source/lintian-overrides to do lintian happy
      about package-needs-versioned-debhelper-build-depends 9.
  * Bump Standards-Version to 3.9.3.
    + Update to DEP5 copyright format 1.0.

 -- Daniel Echeverry <epsilon77@gmail.com>  Sat, 24 Mar 2012 23:51:36 -0500

hydra (7.2-1) unstable; urgency=low

  * New upstream release.
  * Refresh 05_add_option_to_cflag.diff patch offset.
  * Extend copyright holders years.
    + Add missing copyright.		

  [Julián Moreno Patiño]
  * Add 07_remove_troubled_files.diff patch to avoid
    a FTBFS in a second row.
  * Bump debhelper to 8.1.3. (compat level 9).
  * Update 02_using_supported_protocols.diff patch to add
    support of AFP protocol to Linux and KfreeBSD arches.
    + Add libafpclient-dev[!hurd-any] to B-D.

 -- Daniel Echeverry <epsilon77@gmail.com>  Thu, 16 Feb 2012 23:32:00 -0500

hydra (7.1-2) unstable; urgency=low

  * Switch compat level 8 to 9 to use hardening flags.
  * Update 05_add_option_to_cflag.diff patch
    to use properly Hardening flags.
  * Add 06_show_xhydra_build_output.diff patch
    to show verbose build mode in xhydra.
  * Remove unnecessary newline in debian/rules.
  * Use firebird-dev instead firebird2.5-dev in B-D. (Closes: #653377)
  * Add lintian override to fix
    package-needs-versioned-debhelper-build-depends 
    warning for experimental debhelper version.

  [Ludovico Cavedon]
  * Add DM-Upload-Allowed.

 -- Julián Moreno Patiño <darkjunix@gmail.com>  Sat, 31 Dec 2011 12:42:34 -0500

hydra (7.1-1) unstable; urgency=low

  * New upstream release
  * Drop 06_fix_manpage.diff
    + Merge with upstream
  * Refresh 02_using_supported_protocols.diff patch
  * Update 04_fix_spelling_error.diff patch
    + Some parts merge with upstream
    + Fix typo

dpl4hydra.sh
#!/system/bin/sh
#
# Name:     dpl4hydra
# Version:  0.9.9
# Date:     2012-04-16
# Author:   Roland Kessler / Twitter: @rokessler
# Synopsis: Generates a (d)efault (p)assword (l)ist as input for THC hydra.
# Credits:  Thanks to van Hauser for support and fixing portability issues.
#           "The universe is an intelligence test." -Timothy Leary (R.I.P.)

INSTALLDIR=$HOME
LOCATION=.dpl4hydra

usage ()
{
cat <<EOF
dpl4hydra v0.9.9 (c) 2012 by Roland Kessler (@rokessler)

Syntax: `basename $0` [help] | [refresh] | [BRAND] | [all]

This script depends on a local (d)efault (p)assword (l)ist called
${FULLFILE}. If it is not available, regenerate it with
'`basename $0` refresh'. Source of the default password list is
$SITE

Options:
  help        Help: Show this message
  refresh     Refresh list: Download the full (d)efault (p)assword (l)ist
              and generate a new local ${FULLFILE} file. Takes time!
  BRAND       Generates a (d)efault (p)assword (l)ist from the local file
              ${FULLFILE}, limiting the output to BRAND systems, using
              the format username:password (as required by THC hydra).
              The output file is called dpl4hydra_BRAND.lst.
  all         Dump list of all systems credentials into dpl4hydra_all.lst.

Example:
# `basename $0` linksys
File dpl4hydra_linksys.lst was created with 20 entries.
# hydra -C ./dpl4hydra_linksys.lst -t 1 192.168.1.1 http-get /index.asp
EOF
}

refresh ()
{
  echo
  echo "Trying to locate wget or curl... " | tr -d "\n"
  which wget >/dev/null 2>&1 && FETCH="wget -q -O -"
  which curl >/dev/null 2>&1 && FETCH="curl -s"

  if [ -n "$FETCH" ]; then
    echo "done."
    echo "Using `echo $FETCH | cut -d ' ' -f 1` for downloading data."
    echo
  else
    echo
    echo "ERROR: Cannot refresh the list without wget or curl. Aborting." >&2
    echo
    exit 1
  fi
  
  echo "Trying to download list of vendors from"
  echo "${SITE}... " | tr -d "\n"
  $FETCH $SITE > $INDEXSITE 2>/dev/null || { echo; echo; echo "ERROR: Downloading data to disk failed. Network down?" >&2; echo; rm $INDEXSITE; exit 1; }
  echo "done."
  echo
  
  cat $INDEXSITE | grep td | awk -F"\"" '{ print $8 }' | grep http > $SUBSITES
  rm $INDEXSITE
  
  if [ -r $FULLFILE ]; then
    echo "Moving existing password list to ${OLDFILE}."
    echo
    mv $FULLFILE $OLDFILE || { echo "ERROR: Moving file $FULLFILE failed. Please check." >&2; echo; exit 1; }
  fi
    
  for SUBSITE in `cat $SUBSITES`; do
    VENDOR=`echo $SUBSITE | awk -F"-" '{ print $3 }' | sed 's/.htm//'`
    echo "Downloading default passwords for ${VENDOR}... " | tr -d "\n"
    $FETCH $SUBSITE | grep -i tr | grep -i td | grep -i celltext | sed 's/<[^>]*>/,/g' | sed 's/,,*/,/g' | sed 's/^,//g' | tr -d "\r" >dpl4hydra_${VENDOR}.tmp || { echo "not found - skipping... " | tr -d "\n" ; }
    
    while read SYSTEM; do
      echo "${VENDOR}," | tr -d "\n" >> $FULLFILE
      echo "$SYSTEM" >> $FULLFILE
    done < dpl4hydra_${VENDOR}.tmp
    
    rm dpl4hydra_${VENDOR}.tmp
    echo "done."
  done
  rm $SUBSITES
  
  if [ ! -r $LOCALFILE ]; then
    echo
    echo "ERROR: Cannot access local file ${LOCALFILE}. Skipping." >&2
    echo
  else
    echo
    echo "Merging download with ${LOCALFILE}... " | tr -d "\n"
    cat $LOCALFILE >> $FULLFILE || { echo; echo "ERROR: Merging of $FULLFILE and $LOCALFILE failed. Please check." >&2; echo; exit 1; }
    echo "done."
  fi
  
  echo "Cleaning up and sorting ${FULLFILE}... " | tr -d "\n"
  cat $FULLFILE | sed 's/(null)//g' | sed 's/(Null)//g' | sed 's/(NULL)//g' | sed 's/(blank)//g' | sed 's/(Blank)//g' | sed 's/(BLANK)//g' | sed 's/(none)//g' | sed 's/(None)//g' | sed 's/(NONE)//g' | sed 's/none//g' | sed 's/n\/a//g' | sed 's/&lt;/</g' | sed 's/&gt;/>/g' | sed 's/&nbsp;//g' | sort | uniq > $CLEANFILE
  mv $CLEANFILE $FULLFILE
  echo "done."
  echo
  echo "Refreshed (d)efault (p)assword (l)ist $FULLFILE"
  echo "was created with `wc -l $FULLFILE | awk '{ print $1 }'` entries."
  echo
}

generate ()
{
  HYDRAFILE=`echo "dpl4hydra_${BRAND}.lst" | tr '/ =:@\\|;<>"'"'" '_____________'`

  if [ ! -r $FULLFILE ]; then
    echo
    echo "ERROR: Cannot access input file ${FULLFILE}" >&2
    echo "       You can rebuild it with '`basename $0` refresh'." >&2
    echo
    echo "       Trying to use $LOCALFILE instead... " | tr -d "\n"
    if [ -r $LOCALFILE ]; then
      FULLFILE=$LOCALFILE
      echo "done."
    else
      echo
      echo "ERROR: Cannot access local file ${LOCALFILE}. Aborting." >&2
      echo
      exit 1
    fi
  fi

  cat $FULLFILE 2>/dev/null | grep -i "$PATTERN" | awk -F"," '{ print $5":"$6 }' | sed 's/^[ \t]*//' | sed 's/[ \t]*$//' | sort | uniq > $DPLPATH/$HYDRAFILE

  ENTRIES=`wc -l $DPLPATH/$HYDRAFILE | awk '{ print $1 }'`
  if [ "$ENTRIES" -eq 0 ]; then
    rm -f $DPLPATH/$HYDRAFILE
    echo
    echo "ERROR: No matching entries found for $BRAND systems." >&2
    echo "       File $HYDRAFILE was not created." >&2
    echo
    exit 1
  else
    if [ "$ENTRIES" -eq 1 ]; then
      echo
      echo "File $HYDRAFILE was created with one entry."
      echo
    else
      echo
      echo "File $HYDRAFILE was created with $ENTRIES entries."
      echo
    fi
  fi
}

LC_ALL=C
export LC_ALL
DPLPATH="."
CSVPATH="/usr/share/hydra"
test -r "$DPLPATH/dpl4hydra_full.csv" || DPLPATH="$INSTALLDIR/$LOCATION"
test -d "$INSTALLDIR/$LOCATION" || mkdir -p "$INSTALLDIR/$LOCATION"
FULLFILE="$DPLPATH/dpl4hydra_full.csv"
OLDFILE="$DPLPATH/dpl4hydra_full.old"
LOCALFILE="$CSVPATH/dpl4hydra_local.csv"
INDEXSITE="$DPLPATH/dpl4hydra_index.tmp"
SUBSITES="$DPLPATH/dpl4hydra_subs.tmp"
CLEANFILE="$DPLPATH/dpl4hydra_clean.tmp"
SITE="http://open-sez.me/passwd.htm"

case $# in
	0) usage
	   exit 0;;
	1) OPT=`echo $1 | tr "[A-Z]" "[a-z]"`;;
	*) echo
     echo "ERROR: Too many options." >&2
     usage
     exit 1;;
esac

case "$OPT" in
  "-h" | "help" | "-help" | "--help")          usage;;
  "-r" | "refresh" | "-refresh" | "--refresh") refresh;;
  "-a" | "all" | "-all" | "--all")             PATTERN=","
                                               BRAND="all"
                                               generate;;
  *)                                           PATTERN="${OPT}"
                                               BRAND="$OPT"
                                               generate;;
esac
